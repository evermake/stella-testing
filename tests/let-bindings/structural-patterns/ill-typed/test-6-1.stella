language core;

extend with #let-bindings, #let-patterns, #structural-patterns, #records;

fn main(n : Nat) -> Nat {
  return
  	let x =
      { a = 0
      , b = true
      , c = fn(x : Bool){ return if x then false else true}
      } in
    let y =
      { a = 0
      , b = true
      , c = fn(x : Bool){ return if x then false else true}
      } in
    let { a = u, d = v, c = z} = x in
    if z(y.b) then x.a else succ(x.a)
}
