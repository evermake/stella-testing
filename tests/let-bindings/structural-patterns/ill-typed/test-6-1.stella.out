An error occurred during typechecking!
Type Error Tag: [ERROR_UNEXPECTED_PATTERN_FOR_TYPE]
unexpected pattern
  
{
  a = u, d = v, c = z
}

when pattern matching is expected for type
  
{
  a : Nat, b : Bool, c : fn (Bool) -> Bool
}